src/main/java/cn/nukkit/command/Command.java

```diff
@@ -1,71 +1,34 @@
 package cn.nukkit.command;
 
 import cn.nukkit.command.data.CommandData;
-import cn.nukkit.command.data.CommandOverload;
-import cn.nukkit.command.data.CommandParamType;
 import cn.nukkit.command.data.CommandParameter;
 import cn.nukkit.locale.TranslationContainer;
 import cn.nukkit.player.Player;
+import cn.nukkit.registry.CommandRegistry;
 import cn.nukkit.utils.TextFormat;
 import co.aikar.timings.Timing;
 import co.aikar.timings.Timings;
-import com.nukkitx.protocol.bedrock.data.CommandEnumData;
-import com.nukkitx.protocol.bedrock.data.CommandParamData;
 
-import java.util.*;
+import java.util.List;
 
 /**
- * author: MagicDroidX
- * Nukkit Project
+ * Base class for Commands. Plugins should extend {@link PluginCommand PluginCommand<T>} and not this class.
+ *
+ * @author MagicDroidX
+ * @see PluginCommand
  */
 public abstract class Command {
 
-    protected CommandData commandData;
+    protected final CommandData commandData;
 
     private final String name;
 
-    private String[] aliases;
-
-    protected String description;
-
-    protected String usageMessage;
-
-    private String permission = null;
-
-    private String permissionMessage = null;
-
-    protected List<CommandParameter[]> commandParameters = new ArrayList<>();
-
     public Timing timing;
 
-    public Command(String name) {
-        this(name, "", null, new String[0]);
-    }
-
-    public Command(String name, String description) {
-        this(name, description, null, new String[0]);
-    }
-
-    public Command(String name, String description, String usageMessage) {
-        this(name, description, usageMessage, new String[0]);
-    }
-
-    public Command(String name, String description, String usageMessage, String[] aliases) {
-        this.commandData = new CommandData();
-        this.name = name.toLowerCase(); // Uppercase letters crash the client?!?
-        this.description = description;
-        this.usageMessage = usageMessage == null ? "/" + name : usageMessage;
-        this.aliases = aliases;
+    public Command(String name, CommandData data) {
```

Comment:
`CommandData` already has the name (`cmdName` field), `name` parameter seems redundant unless there's some special reason i'm not aware of. at any rate, an overload which accepts only a `CommandData` would be handy

Reply:
Agreed. It was mostly left over from when I was doing CommandFactory based off name.